### PR TITLE
Add docs for few CMake variables

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Fixes https://github.com/STEllAR-GROUP/hpx/issues/1258

## Proposed Changes
Added documentation for listed CMake variables in the ticket.

## Any background context you want to provide?
GSoD'19 applicant.

### CODE REVIEW

File: docs/sphinx/manual/building_hpx.rst [hunk 1]
```diff
@@ -1357,3 +1357,21 @@ or Unix variants.
...
+This lists the CMAKE variables and how user can enable CMAKE variables during
+make.
+
+- `PLUGIN` : This component is a plugin-able library.
+- `TYPE` : The type can be: EXECUTABLE, LIBRARY or COMPONENT.
+- `INSTALL`: Generates a install rule for the target.
+- `HPX_SetOutputPaths.cmake`: If you want the libs in directory where you can
+  add_executable, you can use `*_OUTPUT_DIRECTORY` variable. To mimic HPX's
+  behavior, set ``include(HPX_SetOutputPaths``) after ``find_package(HPX)``.
+- `LD_LIBRARY_PATH` : In |hpx| libraries are in a path that isn't found by the
```
Comment (hkaiser) [inline docs/sphinx/manual/building_hpx.rst]: `LD-LIBRARY_PATH` is not a variable related to cmake but only usable during execution of the application. It also is specific to Linux.

File: docs/sphinx/manual/building_hpx.rst [hunk 2]
```diff
@@ -1357,3 +1357,21 @@ or Unix variants.
...
+
+CMAKE Variable
+==============
+
+This lists the CMAKE variables and how user can enable CMAKE variables during
+make.
+
+- `PLUGIN` : This component is a plugin-able library.
+- `TYPE` : The type can be: EXECUTABLE, LIBRARY or COMPONENT.
+- `INSTALL`: Generates a install rule for the target.
```
Comment (hkaiser) [inline docs/sphinx/manual/building_hpx.rst]: These are not 'cmake variables' (whatever this means). Using them on the cmake command line will have no effect. The important variables that can be passed in the form of `-DSOME_VARIABLE=<value>` on cmake's command line are listed in the generated file `generated/cmake_variables.rst`. 

What was your rationale to add this text?

File: docs/sphinx/manual/building_hpx.rst (hunk 1 above)
Comment (tapaswenipathak) [reply docs/sphinx/manual/building_hpx.rst]: Can you write what should be correct heading ref https://github.com/STEllAR-GROUP/hpx/issues/1258#issuecomment-55564243.

File: docs/sphinx/manual/building_hpx.rst (hunk 2 above)
Comment (tapaswenipathak) [reply docs/sphinx/manual/building_hpx.rst]: Aren't these `hpx_setup_target(target)` macro parameters? applicable on https://github.com/STEllAR-GROUP/hpx/issues/1258#issue-42727647

Comment (msimberg) [reply docs/sphinx/manual/building_hpx.rst]: @tapaswenipathak I'd say these are already documented [here](https://stellar-group.github.io/hpx/docs/sphinx/branches/master/html/manual/creating_hpx_projects.html#cmake-macros-to-integrate-hpx-into-existing-applications). Would you agree?

Comment (tapaswenipathak) [reply docs/sphinx/manual/building_hpx.rst]: Sure, can you tell me what should be documented for https://github.com/STEllAR-GROUP/hpx/issues/1258?

Comment (msimberg) [reply docs/sphinx/manual/building_hpx.rst]: @tapaswenipathak see my comment on the issue.

Comment (tapaswenipathak) [reply docs/sphinx/manual/building_hpx.rst]: Reading your comment on the issue I think  we should close this PR @msimberg?

**Comment by msimberg** [comment]
@tapaswenipathak yeah, going to close this now. If you're still interested in working in this direction I'd happily talk to you on IRC or Slack to decide on something more substantial to work on.

**Pull request closed without merging**